### PR TITLE
Adding a model consistency unit test and fixing the uncovered discrepancies

### PR DESCRIPTION
**Pull request by zaccharieramzi**
Added a model consistency test to check that we have indeed the same models, and the same optimizers.

Also added a script to check the speed between the 2 frameworks (not really telling because the difference is really high so there might be something to check there).

The model consistency check led to 3 corrections in model implementation:
- corrected the padding in downsampling modules of the tf resnets to match that of PyTorch (non-critical). Note that this might cause a slight overhead for the TF implementation (as noted [here](https://github.com/keras-team/keras/pull/16363#issuecomment-1099699173)), so we can chose to remove it now that we figured out the other criticial differences
- removed an extra ReLU in the first layer of the ResNet configured for CIFAR (translation of the model configured for ImageNet) in the tf implem (critical a priori)
- corrected the padding in the first layer of the ResNet configured for CIFAR in the torch implem (critical a priori, esp. for computational efficiency)

Also increased patience, cos why not.

### CODE REVIEW

**Comment by zaccharieramzi** [comment]
The current failure is due to a failed http request when downloading CIFAR. Otherwise the tests are all green (confirmed in local).

**Comment by zaccharieramzi** [comment]
not waiting for the macos workers
